cmd/maintained-apps/validate/main.go

```diff
@@ -0,0 +1,326 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"slices"
+	"strings"
+	"time"
+
+	maintained_apps "github.com/fleetdm/fleet/v4/ee/maintained-apps"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	mdm_maintained_apps "github.com/fleetdm/fleet/v4/server/mdm/maintainedapps"
+)
+
+var (
+	tmpDir                      string
+	env                         []string
+	installationSearchDirectory string
+	operatingSystem             string
+)
+
+func run() int {
```

Comment:
style thing: I'd advocate for returning an error here, then logging and returning an error code in `main`.